### PR TITLE
[Refactor] Refactor all the GlobalCommandOptions from pointer to value reference

### PR DESCRIPTION
**Pull request by Zheaoli**
Based on the discussion https://github.com/containerd/nerdctl/pull/1820#discussion_r1064666038

Signed-off-by: Zheao.Li <me@manjusaka.me>

### CODE REVIEW

File: cmd/nerdctl/volume.go [hunk 1]
```diff
@@ -44,6 +44,6 @@ func newVolumeCommand() *cobra.Command {
 
 // getVolumeStore returns a volume store
 // that corresponds to a directory like `/var/lib/nerdctl/1935db59/volumes/default`
-func getVolumeStore(globalOptions *types.GlobalCommandOptions) (volumestore.VolumeStore, error) {
+func getVolumeStore(globalOptions types.GlobalCommandOptions) (volumestore.VolumeStore, error) {
 	return volume.Store(globalOptions.Namespace, globalOptions.DataRoot, globalOptions.Address)
 }
```
Comment (djdongjin) [inline cmd/nerdctl/volume.go:49]: I suggest we remove this function and use `volume.Store` (the 2nd one below) directly (we can change `volume.Store` to accept a globalOptions) (Don't necessarily be part of this PR though)

This adds one unnecessary wrap on volume create:

https://github.com/containerd/nerdctl/blob/40d341dac245a70a6ee4ac43567f2d2818fe3167/cmd/nerdctl/volume.go#L47-L49

https://github.com/containerd/nerdctl/blob/40d341dac245a70a6ee4ac43567f2d2818fe3167/pkg/cmd/volume/volume.go#L26-L32

https://github.com/containerd/nerdctl/blob/40d341dac245a70a6ee4ac43567f2d2818fe3167/pkg/mountutil/volumestore/volumestore.go#L42

File: cmd/nerdctl/run.go [hunk 2]
```diff
@@ -395,7 +395,7 @@ func runAction(cmd *cobra.Command, args []string) error {
 }
 
 // FIXME: split to smaller functions
-func createContainer(ctx context.Context, cmd *cobra.Command, globalOptions *types.GlobalCommandOptions, client *containerd.Client, args []string, platform string, flagI, flagT, flagD bool) (containerd.Container, func(), error) {
+func createContainer(ctx context.Context, cmd *cobra.Command, globalOptions types.GlobalCommandOptions, client *containerd.Client, args []string, platform string, flagI, flagT, flagD bool) (containerd.Container, func(), error) {
```
Comment (djdongjin) [inline cmd/nerdctl/run.go:398]: nit: some functions have `containerd.Client` argument before `options` and some after `options`. E.g.:

```golang
func generateRootfsOpts(ctx context.Context, client *containerd.Client, platform string, cmd *cobra.Command, globalOptions types.GlobalCommandOptions, args []string, id string) ([]oci.SpecOpts, []containerd.NewContainerOpts, *imgutil.EnsuredImage, error) {
```


We might want to keep consistent style (personally I think functional arguments like `client`, `container`, `image` should be before `options`).

**Review by djdongjin — COMMENTED** [review]
Overall LGTM, left some minor comments. thanks

**Review by djdongjin — APPROVED** [review]

File: cmd/nerdctl/volume.go (hunk 1 above)
Comment (Zheaoli) [reply cmd/nerdctl/volume.go:49]: This should be another PR

File: cmd/nerdctl/run.go (hunk 2 above)
Comment (Zheaoli) [reply cmd/nerdctl/run.go:398]: SGTM, I will update it

**Review by djdongjin — APPROVED** [review]
LGTM on CI green. thanks

**Pull request merged**